src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1952,6 +1928,16 @@ private TypeParameterConstraintClauseSyntax ParseTypeParameterConstraintClause()
                 name,
                 colon,
                 _pool.ToListAndFree(bounds));
+
+            PostSkipAction skipBadTypeParameterConstraintTokens(SeparatedSyntaxListBuilder<TypeParameterConstraintSyntax> list, SyntaxKind expected)
+            {
+                CSharpSyntaxNode tmp = null;
+                Debug.Assert(list.Count > 0);
+                return this.SkipBadSeparatedListTokensWithExpectedKind(ref tmp, list,
+                    p => p.CurrentToken.Kind != SyntaxKind.CommaToken && !p.IsPossibleTypeParameterConstraint(),
+                    p => p.CurrentToken.Kind == SyntaxKind.OpenBraceToken || p.IsCurrentTokenWhereOfConstraintClause() || p.IsTerminator(),
+                    expected);
+            }
```

Comment:
moving the skip helpers inside prevents drift where other helpers outside the method push tehse further away.